### ISSUE TITLE
prefer_final_fields false positive if field is set from another parts

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#2697]
```dart
//----- lib.dart
part 'a.dart';
part 'b.dart';


//----- a.dart
part of 'lib.dart';
class A {
  A(this._o);
  String _o; // false positive prefer_final_fields
}


//----- b.dart
part of 'lib.dart';
f(A a) => a._o = '';
```

**Comment by pq** [comment]
As per https://github.com/dart-lang/linter/issues/4992#issuecomment-2326988151, closing as fixed w/ https://github.com/dart-lang/linter/commit/ad42f65d694bcc82169fcbdf2d8bdbce7ba4d34d#diff-54e392d799cbae383257d19e489ee7c578d6c181414e70a037d810e978c93532
